common-service-core/src/main/java/com/ca/mfaas/message/api/ApiMessage.java

```diff
@@ -20,12 +24,36 @@
  * However remember that one kind of user may be a person with malicious intent so not leak data that should be kept
  * private or implementation details.
  */
+@Setter
+@EqualsAndHashCode
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public interface Message {
+public class ApiMessage<T> {
+
+    private MessageType messageType;
+    private String messageNumber;
+    private String messageContent;
+    private String messageAction;
+    private String messageReason;
+    private String messageKey;
+    private List<T> messageParameters;
+    private String messageInstanceId;
+    private String messageComponent;
+    private String messageSource;
+
+
+    public ApiMessage(String messageKey, MessageType messageType, String messageNumber, String messageContent) {
+        this.messageKey = messageKey;
+        this.messageType = messageType;
+        this.messageNumber = messageNumber;
+        this.messageContent = messageContent;
+    }
+
     /**
      * The severity of a problem. This field is required.
      */
-    MessageType getMessageType();
+    public MessageType getMessageType() {
+        return messageType;
+    }
 
     /**
      * Typical mainframe message number (not including the message level one-letter code) that can be found in CA
```

Comment:
"Typical mainframe message number (excluding the 'level one-letter code') that can be found in CA"
Now, we should probably change 'CA' to 'Broadcom', or be more specific re what "CA" refers to.

Reply:
We didn't change this message, not sure what CA refers about. We can change to Broadcom anyway

Reply:
right, then maybe we shouldn't touch it before we figure what is meant by 'CA'